### ISSUE TITLE
Constant input warning for PearsonR

### ISSUE THREAD

**Issue by FrancescaDr** [theislab/interscale#19]
### Report

GlobalModel training gives Constant input warning for PearsonR (without masking).

- [x] Find the cause: Is y_pred or y_true constant?

```
GT-long-range-niches/docs/notebooks/1-Projects/GT-long-range-niches/src/InterScale/train/_trainingplans.py#line=180): ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.
  pearson_corr_raw = pearsonr(y_pred_np, y_true_np)
```

**Comment by FrancescaDr** [comment]
The error occurs if one axis has no variation. There are two possible reasons for this:

1. cross-cell per gene: A gene is not expressed in any of the cells due to sparsity and sub-selection of cells for transformer sequence length. The pearson correlation score is not defined for that gene but not considering NaN we still get an estimation, but not sure how reliable those are. When we are using np.nanmean() to summarize the predicted values then it gives an average ignoring the NaNs. 
2. cross-gene per cell: A cell has zero one expression -> need to filter out zero expression cells before running the model.

```
# Example: Gene A is not expressed
y_pred_np = torch.tensor([[0.1, 0.2, 0.3], [0.15, 0.25, 0.35], [0.7, 0.4, 0.6], [0.4, 0.5, 0.6]])
y_true_np = torch.tensor([[0.0, 0.4, 0.6], [0.0, 0.34, 0.43], [0.0, 0.4, 0.3], [0.0, 0.2, 0.3]])

#cross-cell per gene
pearson_corr_raw = pearsonr(y_pred_np, y_true_np)
pearson_corr = torch.tensor(np.nanmean(pearson_corr_raw[0]), dtype=torch.float32, device=y_pred_np.device)
print(pearson_corr_raw, pearson_corr)

#cross-gene per cell
pearson_corr_raw = pearsonr(y_pred_np.T, y_true_np.T)
pearson_corr = torch.tensor(np.nanmean(pearson_corr_raw[0]), dtype=torch.float32, device=y_pred_np.device)
print(pearson_corr_raw, pearson_corr)

----
OUTPUT:
PearsonRResult(statistic=array([ nan, 0.27735016, -0.902933 ], dtype=float32), pvalue=array([ nan, 0.82108757, 0.28281871]))
tensor(-0.3128)

# cross-gene per cell
PearsonRResult(statistic=array([ 0.9819805 ,  0.94801688, -0.89104206,  0.98198053]), pvalue=array([0.12103773, 0.20617016, 0.29995015, 0.12103763], dtype=float32)) tensor(0.5052)
